examples/stm32h7-example/Core/Src/cli_setup.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * cli_setup.c
+ *
+ *  Created on: Jul 7, 2023
+ *      Author: NeusAap
+ */
+
+
+// STD LIBS
+#include <stdio.h>
+#include <stdlib.h>
+
+// STM files
+#include "usart.h"
+#include "stm32h7xx_it.h"
+
+// Own CLI headers
+#include "cli_setup.h"
+#include "cli_binding.h"
+// Embedded CLI header (only one allowed per project)
+#define EMBEDDED_CLI_IMPL
+
+// UART buffers
+uint8_t UART_CLI_rxBuffer[UART_RX_BUFF_SIZE] = {0};
+struct {
+    char buffer[UART_TX_BUFFER_SIZE];
+    uint8_t index;
+} UART_CLI_txBuffer = { .index = 0 };
+
+// CLI buffer
+EmbeddedCli *cli;
+CLI_UINT cliBuffer[BYTES_TO_CLI_UINTS(CLI_BUFFER_SIZE)];
+
+// Bool to disable the interrupts, if CLI is not yet ready.
+bool cliIsReady = false;
```

Comment:
Is this used anywhere? Didn't find here or in reference implementation from your repository. If not it should be static as other variables. If it is used, it's better to make function declaration in header and in definition return this variable. In this case it also should be static.

Reply:
It is only used locally, to enable the UART_Rx_Callback, only after initialization of the CLI is finished (`line 91`, in function: `HAL_UART_RxCpltCallback()`). I've changed the variable declaration to static. 

Since it is only used locally in that specific source file, what need to add it to the header file? Making it static should be enough right?

Reply:
Yes, static is the right choice here.